test/e2e/retina_e2e_test.go

```diff
@@ -0,0 +1,61 @@
+package retina
+
+import (
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/microsoft/retina/test/e2e/common"
+	"github.com/microsoft/retina/test/e2e/framework/types"
+	jobs "github.com/microsoft/retina/test/e2e/jobs"
+	"github.com/stretchr/testify/require"
+)
+
+// TestE2ERetina tests all e2e scenarios for retina
+func TestE2ERetina(t *testing.T) {
+	curuser, err := user.Current()
+	require.NoError(t, err)
+
+	clusterName := curuser.Username + common.NetObsRGtag + strconv.FormatInt(time.Now().Unix(), 10)
+
+	subID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	require.NotEmpty(t, subID)
+
+	location := os.Getenv("AZURE_LOCATION")
+	if location == "" {
+		location = "eastus"
+	}
+
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+
+	// Get to root of the repo by going up two directories
+	rootDir := filepath.Dir(filepath.Dir(cwd))
+
+	chartPath := filepath.Join(rootDir, "deploy", "manifests", "controller", "helm", "retina")
+	profilePath := filepath.Join(rootDir, "test", "profiles", "advanced", "values.yaml")
+	kubeConfigFilePath := filepath.Join(rootDir, "test", "e2e", "test.pem")
+
+	// CreateTestInfra
+	createTestInfra := types.NewRunner(t, jobs.CreateTestInfra(subID, clusterName, location, kubeConfigFilePath))
+	createTestInfra.Run()
+
+	// Hacky way to ensure that the test infra is deleted even if the test panics
+	defer func() {
+		if r := recover(); r != nil {
+			t.Logf("Recovered in TestE2ERetina, %v", r)
```

Comment:
This will be printed, but the infra delete won't happen since it's outside of the recover block. 
Don't we also want to delete when test fails to avoid running out of aks resources?